draft-ietf-tls-external-psk-importer.md

```diff
@@ -79,156 +80,151 @@ when, and only when, they appear in all capitals, as shown here.
 PSKs for TLS 1.2. Indeed, this is necessary for incremental deployment.
 
 # Security Considerations
 
-This is a WIP draft and has not yet seen significant security analysis.
+DISCLAIMER: This is a WIP draft and has not yet seen significant security analysis.
 
 # Privacy Considerations
 
-DISCLAIMER: This section contains a sketch of a design for protecting external PSK identities.
-It is not meant to be implementable as written.
-
 External PSK identities are typically static by design so that endpoints may use them to
-lookup keying material. For some systems and use cases, this identity may become a persistent
-tracking identifier. One mitigation to this problem is encryption. Future drafts may specify
-a way for encrypting PSK identities using a mechanism similar to that of the Encrypted
-SNI proposal {{?I-D.ietf-tls-esni}}. Another approach is to replace the identity with an
-unpredictable or "obfuscated" value derived from the corresponding PSK. One such proposal, derived
-from a design outlined in {{?I-D.ietf-dnssd-privacy}}, is as follows. Let ipskx be the imported
-PSK with identity ImportedIdentity, and N be a unique nonce of length equal to that of ImportedIdentity.hash.
-With these values, construct the following "obfuscated" identity:
+lookup keying material. However, for some systems and use cases, this identity may become a
+persistent tracking identifier.
 
-~~~
-   struct {
-       opaque nonce[hash.length];
-       opaque obfuscated_identity<1..2^16-1>;
-       HashAlgorithm hash;
-   } ObfuscatedIdentity;
-~~~
+# IANA Considerations {#IANA}
 
-ObfuscatedIdentity.nonce carries N, ObfuscatedIdentity.obfuscated_identity carries HMAC(ipskx, N),
-where HMAC is computed with ImportedIdentity.hash, and ObfuscatedIdentity.hash is ImportedIdentity.hash.
+This specification introduces a new registry for TLS KDF identifiers and defines the following
+KDFID values:
 
-Upon receipt of such an obfuscated identity, a peer must lookup the corresponding PSK by exhaustively
-trying to compute ObfuscatedIdentity.obfuscated_identity using ObfuscatedIdentity.nonce and each of its
-known imported PSKs. If N is chosen in a predictable fashion, e.g., as a timestamp, it may be possible
-for peers to precompute these obfuscated identities to ease the burden of trial decryption.
++--------------------+-------------+
+| Description        | Value       |
++--------------------+-------------+
+| Reserved           | {0x00,0x00} |
+|                    |             |
+| HKDF_SHA256        | {0x00,0x01} |
+|                    |             |
+| HKDF_SHA384        | {0x00,0x02} |
++--------------------+-------------+
 
-# IANA Considerations
+New KDFID values are allocated according to the following process:
 
-This document makes no IANA requests.
+- Values with the first in the range 0-254 (decimal) are assigned via Specification Required {{!RFC8126}}.
+- Values with the first byte 255 (decimal) are reserved for Private Use {{!RFC8126}}.
```

Comment:
You can use ranges: 0x0000-0xfeff and 0xff00-0xffff.

Reply:
Thanks -- will fix!